src/pages/Settings.tsx

```diff
@@ -0,0 +1,215 @@
+import { Box, Paper, Switch, Typography } from "@mui/material";
+import { useContext, useEffect, useState } from "react";
+import UserContext from "../comps/context/UserContext";
+import { supabase } from "../supabaseClient";
+import { enqueueSnackbar } from "notistack";
+import LoginGate from "../comps/ui/LoginGate";
+
+type Memberships = {
+    organization_id: number;
+    allow_notifications: boolean;
+    organization_name?: string;
+};
+
+const Settings = () => {
+    const user = useContext(UserContext);
+    const [memberships, setMemberships] = useState<Memberships[]>([]);
+
+    useEffect(() => {
+        const fetchMemberships = async () => {
+            const { data: membershipsData, error: membershipsError } =
+                await supabase
+                    .from("memberships")
+                    .select(
+                        `
+                    organization_id,
+                    allow_notifications
+                `,
+                    )
+                    .eq("user_id", user.id)
+                    .returns<Memberships[]>();
+
+            if (membershipsError) {
+                enqueueSnackbar("Failed to fetch memberships", {
+                    variant: "error",
+                });
+                return;
+            }
+
+            if (membershipsData && membershipsData.length > 0) {
+                const organizationIds = membershipsData.map(
+                    (m) => m.organization_id,
+                );
+
+                const { data: organizationsData, error: organizationsError } =
+                    await supabase
+                        .from("organizations")
+                        .select("id, name")
+                        .in("id", organizationIds);
+
+                if (organizationsError) {
+                    enqueueSnackbar("Failed to fetch organization names", {
+                        variant: "error",
+                    });
+                    return;
+                }
+
+                const mergedData = membershipsData.map((membership) => {
+                    const organization = organizationsData.find(
+                        (org) => org.id === membership.organization_id,
+                    );
+                    return {
+                        ...membership,
+                        organization_name: organization
+                            ? organization.name
+                            : "Unknown",
+                    };
+                });
+
+                setMemberships(mergedData);
+            }
+        };
+
+        fetchMemberships();
+    }, [user.id]);
+
+    const handleToggle = async (
+        organization_id: number,
+        currentValue: boolean,
+    ) => {
+        const { data, error } = await supabase
+            .from("memberships")
+            .update({ allow_notifications: !currentValue })
```

Comment:
Interesting that this returns 204 even when the change does not happen

Reply:
This is intended behavior https://stackoverflow.com/a/78809858 - perhaps we add a .select() clause and compare the returned output to verify